test/components/connect.spec.js

```diff
@@ -949,6 +949,38 @@ describe('React', () => {
       expect(decorated.refs.wrappedInstance.someInstanceMethod()).toBe(someData)
     })
 
+    it('should do no parsing when the response body is empty', (done) => {
+      window.fetch = (request) => {
+        return new Promise((resolve) => {
+          resolve(new window.Response('', { status: 204 }))
+        })
+      }
+
+      @connect(() => ({ testFetch: `/empty` }))
+      class Container extends Component {
+        render() {
+          return <Passthrough {...this.props} />
+        }
+      }
+
+      const container = TestUtils.renderIntoDocument(
+        <Container />
+      )
+
+      const init = TestUtils.findRenderedComponentWithType(container, Container)
+      expect(init.state.data.testFetch).toIncludeKeyValues(
+        { fulfilled: false, pending: true, reason: null, refreshing: false, rejected: false, settled: false, value: null }
+      )
+
+      setImmediate(() => {
+        const fulfilled = TestUtils.findRenderedComponentWithType(container, Container)
+        expect(fulfilled.state.data.testFetch).toIncludeKeyValues(
+          { fulfilled: true, pending: false, reason: null, refreshing: false, rejected: false, settled: true, value: null }
+        )
+        done()
+      })
+    })
+
```

Comment:
FWIW I just tried running the tests locally with your fix commented, but I see the full object:

```
  1) React connect should not parse the body if response is a 204:
     Uncaught Error: expected { fulfilled: false, meta: { request: { _bodyInit: undefined, _bodyText: '', credentials: 'same-origin', headers: { map: { accept: [Object], 'content-type': [Object] } }, method: 'GET', mode: null, referrer: null, url: '/empty' }, response: { _bodyInit: '', _bodyText: '', bodyUsed: true, headers: { map: { 'content-type': [Object] } }, ok: true, status: 204, statusText: undefined, type: 'default', url: '' } }, pending: false, reason: [SyntaxError: Unexpected end of input], refreshing: false, rejected: true, settled: true, value: null } to include { 'fulfilled': true }
      at Function.assert (node_modules/expect/lib/assert.js:20:9)
      at setup.js:16:14
      at Array.forEach (native)
      at Expectation.toIncludeKeyValues (setup.js:11:27)
      at Immediate._onImmediate (connect.spec.js:977:48)
```